lib/test/t_IntegrationExpansion_std.cxx

```diff
@@ -0,0 +1,172 @@
+ *
+ */
+#include "openturns/OT.hxx"
+#include "openturns/OTtestcode.hxx"
+
+using namespace OT;
+using namespace OT::Test;
+
+int main(int, char *[])
+{
+  TESTPREAMBLE;
+  OStream fullprint(std::cout);
+  setRandomGenerator();
+
+  try
+  {
+
+    // Problem parameters
+    UnsignedInteger dimension = 3;
+    Scalar a = 7.0;
+    Scalar b = 0.1;
+    // Reference analytical values
+    Scalar meanTh = a / 2;
+    Scalar covTh = (pow(b, 2.0) * pow(M_PI, 8.0)) / 18.0 + (b * pow(M_PI, 4.0)) / 5.0 + (pow(a, 2.0)) / 8.0 + 1.0 / 2.0;
+    Point sob_1(3);
+    sob_1[0] = (b * pow(M_PI, 4.0) / 5.0 + pow(b, 2.0) * pow(M_PI, 8.0) / 50.0 + 1.0 / 2.0) / covTh;
+    sob_1[1] = (pow(a, 2.0) / 8.0) / covTh;
+    sob_1[2] = 0.0;
+    Point sob_2(3);
+    sob_2[0] = 0.0;
+    sob_2[1] = (pow(b, 2.0) * pow(M_PI, 8.0) / 18.0 - pow(b, 2.0) * pow(M_PI, 8.0) / 50.0) / covTh;
+    sob_2[2] = 0.0;
+    Point sob_3(1, 0.0);
+    Point sob_T1(3);
+    sob_T1[0] = sob_1[0] + sob_2[0] + sob_2[1] + sob_3[0];
+    sob_T1[1] = sob_1[1] + sob_2[0] + sob_2[2] + sob_3[0];
+    sob_T1[2] = sob_1[2] + sob_2[1] + sob_2[2] + sob_3[0];
+    Point sob_T2(3);
+    sob_T2[0] = sob_2[0] + sob_2[1] + sob_3[0];
+    sob_T2[1] = sob_2[0] + sob_2[2] + sob_3[0];
+    sob_T2[2] = sob_2[1] + sob_2[2] + sob_3[0];
+    Point sob_T3(sob_3);
+    // Create the Ishigami function
+    Description inputVariables(dimension);
+    inputVariables[0] = "xi1";
+    inputVariables[1] = "xi2";
+    inputVariables[2] = "xi3";
+    Description formula(1);
+    formula[0] = (OSS() << "sin(xi1) + (" << a << ") * (sin(xi2)) ^ 2 + (" << b << ") * xi3^4 * sin(xi1)");
+    SymbolicFunction model(inputVariables, formula);
+
+    // Create the input distribution
+    Collection<Distribution> marginals(dimension);
+    marginals[0] = Uniform(-M_PI, M_PI);
+    marginals[1] = Uniform(-M_PI, M_PI);
+    marginals[2] = Uniform(-M_PI, M_PI);
+    ComposedDistribution distribution(marginals);
+
+    // Create the orthogonal basis
+    Collection<OrthogonalUniVariatePolynomialFamily> polynomialCollection(dimension);
+    polynomialCollection[0] = LegendreFactory();
+    polynomialCollection[1] = LegendreFactory();
+    polynomialCollection[2] = LegendreFactory();
+
+    LinearEnumerateFunction enumerateFunction(dimension);
+    OrthogonalProductPolynomialFactory productBasis(polynomialCollection, enumerateFunction);
+
+    // Create the projection strategy
+    UnsignedInteger samplingSize = 250;
+    // Monte Carlo sampling
+    Sample inputSample(LowDiscrepancyExperiment(LowDiscrepancySequence(SobolSequence()), distribution, samplingSize).generate());
+    Sample outputSample(model(inputSample));
+    // From here, the model is no more needed
+    UnsignedInteger degree = 6;
+    UnsignedInteger basisSize = enumerateFunction.getStrataCumulatedCardinal(degree / 2);
+    // Create the polynomial chaos algorithm
+    IntegrationExpansion algo(inputSample, outputSample, distribution, productBasis, basisSize);
```

Comment:
Need to add test with `weights` different from `1`

Reply:
Fixed

Reply:
Where?